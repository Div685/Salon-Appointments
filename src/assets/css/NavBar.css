.header {
  width: 100%;
  text-align: center;
  /* color: rgb(207, 204, 204)207, 204, 204); */
  position: relative;
}

.nav__logo a{
  color: #000;
  font-size: 20px;
}

a {
  color: #000;
  text-decoration: none;
}

.nav span {
  padding: 12px 0;
}

.nav a {
  padding: 12px 0;
}

.nav a:hover {
  background: #97BF12;
  color: #fff;
}

.nav {
  width: 100%;
  display: flex;
  flex-direction: column;
  box-shadow: 1px 1px 5px rgba(#000, 0.2);
  position: absolute;
  background: #fff;
  z-index: 99;
  left: 0;
  transform: translateY(-200%);
  transition: all 0.25s;
}

nav.nav.active {
  width: 100%;
  transform: translateY(62%);
}

.nav__toggle {
    cursor: pointer;
}

.nav__toggle button {
  border: none;
  background: transparent;
}

.nav__logo-tab {
  display: none;
}

.logoutButton {
  background: #97BF12;
    color: #fff;
}

@media only screen 
  and (min-device-width: 768px) 
  and (max-device-width: 1024px) {

    .header {
      /* width: 20%;
      text-align: center;
      position: fixed;
      height: 100%;
      background: #fff;
      left: 0;
      z-index: 999; */
      background: #888;
    height: 100vh;
    padding: 25px 0;
    }
    
    .nav__logo {
      display: block;
    }

    .nav__logo a{
      color: #000;
      font-size: 20px;
    }
    
    a {
      color: #000;
      text-decoration: none;
    }
    
    .nav span {
      padding: 12px 0;
    }
    
    .nav a {
      padding: 12px 0;
      width: 100%;
    display: flex;
    align-items: center;
    justify-content: center
    }
    
    .nav a:hover {
      background: #97BF12;
      color: #fff;
    }
    
    .nav {
      display: flex;
      flex-direction: column;
      background: #fff;
      transform: translateY(200%);
      justify-content: center;
      align-items: center;
      transition: all 0.25s;
    }
    
    nav.nav.active {
      width: 100%;
      transform: translateY(62%);
    }
    
    .nav__toggle {
        cursor: pointer;
        display: none;
    }
    
    .nav__toggle button {
      border: none;
      background: transparent;
    }
    
    .nav__logo-tab {
      display: none;
    }
    
    .logoutButton {
      background: #97BF12;
        color: #fff;
    }
}