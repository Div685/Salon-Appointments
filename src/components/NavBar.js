import React, { useState } from 'react';
import { NavLink, Link } from 'react-router-dom';
import MenuIcon from '@material-ui/icons/Menu';
import CancelIcon from '@material-ui/icons/Cancel';
import { useSelector } from 'react-redux';
import LogOutBtn from './LogOutBtn';

const NavBar = () => {
  const [toggleNavClass, setToggleNavClass] = useState('');
  const [isClicked, setIsClicked] = useState(true);

  const isLoggedIn = useSelector((state) => state.user.logIn);
  const user = useSelector((state) => state.user);

  const changeButtonName = () => {
    setIsClicked(!isClicked);
  };

  const handleToggle = () => {
    setToggleNavClass(toggleNavClass === ' active' ? '' : ' active');
    changeButtonName();
  };

  const closeToggle = () => {
    setToggleNavClass('');
    changeButtonName();
  };

  return (
    <header className="header">
      <div className="nav__logo">
        <Link to="/">
          <h3>BookMySalon</h3>
        </Link>
      </div>
      <div className="nav__toggle">
        <button type="button" className={`header_button${toggleNavClass}`} onClick={handleToggle}>
          { isClicked ? (
            <MenuIcon />
          ) : (<CancelIcon />) }
        </button>
      </div>
      <nav className="nav">
        {isLoggedIn ? (
          <div className={`nav__items${toggleNavClass}`}>
            <span>{`Welcome ${user.user.username}`}</span>
            <NavLink exact to="/items" activeClassName="selected" onClick={closeToggle}>Items</NavLink>
            <NavLink exact to="/appointments" activeClassName="selected" onClick={closeToggle}>Appointments</NavLink>
            <NavLink exact to="/" activeClassName="selected" onClick={closeToggle}>LifeStyle</NavLink>
            <LogOutBtn />
          </div>
        ) : (
          <div className={`nav__auth${toggleNavClass}`}>
            <NavLink exact to="/login" activeClassName="selected" onClick={closeToggle}>Login</NavLink>
            <NavLink exact to="/signup" activeClassName="selected" onClick={closeToggle}>Sign Up</NavLink>
          </div>
        ) }
      </nav>
    </header>
  );
};

NavBar.defaultProps = {
  user: {},
};

export default NavBar;
