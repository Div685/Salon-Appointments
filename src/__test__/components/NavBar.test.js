import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NavBar from '../../components/NavBar';
import '@testing-library/jest-dom';
import configureStore from '../helper/createStore';

let store;

describe('NavBar Component', () => {
  beforeEach(() => {
    store = configureStore();
  });
  it('renders NavBar page correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('Test for NavBar component', () => {
  it('test the heading of nav component', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>
      </Provider>,
    );
    expect(screen.getByRole('heading')).toHaveTextContent('BookMySalon');
  });
});
