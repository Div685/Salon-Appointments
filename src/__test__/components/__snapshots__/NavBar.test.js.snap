// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders NavBar page correctly 1`] = `
<header
  className="header"
>
  <div
    className="nav__logo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h3>
        BookMySalon
      </h3>
    </a>
  </div>
  <div
    className="nav__toggle"
  >
    <button
      className="header_button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </div>
  <nav
    className="nav"
  >
    <div
      className="nav__items"
    >
      <span>
        Welcome undefined
      </span>
      <a
        aria-current={null}
        href="/items"
        onClick={[Function]}
      >
        Items
      </a>
      <a
        aria-current={null}
        href="/appointments"
        onClick={[Function]}
      >
        Appointments
      </a>
      <a
        aria-current="page"
        className="selected"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        LifeStyle
      </a>
      <button
        className="btn logoutButton"
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </div>
  </nav>
</header>
`;
